services/UserAccountService.ts

```diff
@@ -166,6 +166,24 @@ export default class UserAccountService {
     });
   }
 
+  public async collectOnboarding(user: UserModel): Promise<UserModel> {
+    const userProfile = await this.getFullUserProfile(user);
+    if (userProfile.attendanceCount < 5
+      || userProfile.resumes.length < 1
+      || userProfile.profilePicture == null
+      || userProfile.bio == null) {
+      throw new BadRequestError('Onboarding tasks not completed!');
+    }
+    if (userProfile.onboardingCollected) {
+      throw new BadRequestError('Onboarding reward already collected!');
+    }
+    return this.transactions.readWrite(async (txn) => {
+      const userRepository = Repositories.user(txn);
+      await userRepository.addPoints(user, 10);
+      return userRepository.upsertUser(user, { onboardingCollected: true });
```

Comment:
Yes, the intention is that when you have all the tasks completed it will light up a button in the profile page to collect the rewards. We felt it was simpler to implement this way and more satisfying to actually get to collect the reward rather than automatically detecting it and then updating it.